### PR TITLE
Tlg tern

### PR DESCRIPTION
**Pull request by pawelru**
add demographic and adverse event tables

interactive part needs to wait as it is under heavy refactor right now

### CODE REVIEW

File: tlg/demographic.qmd [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+---
+
+## Introduction
+
+This guide will show you how pharmaverse packages, along with some from tidyverse, can be used to create a Demographic table, using the `{pharmaverseadam}` `ADSL` data as an input.
+
+The packages used with a brief description of their purpose are as follows:
+
+* [`{rtables}`](https://roche.github.io/rtables/index.html): designed to create and display complex tables with R.
+* [`{tern}`](https://insightsengineering.github.io/tern/main/index.html): contains analysis functions to create tables and graphs used for clinical trial reporting.
```
Comment (pawelru) [inline tlg/demographic.qmd:13]: Fix links

```suggestion
* [`{rtables}`](https://insightsengineering.github.io/rtables/): designed to create and display complex tables with R.
* [`{tern}`](https://insightsengineering.github.io/tern/): contains analysis functions to create tables and graphs used for clinical trial reporting.
```

File: DESCRIPTION [hunk 2]
```diff
@@ -9,16 +9,19 @@ Description: This is not a package, but we just use this file to declare
...
+    admiral,
     dplyr,
+    lubridate,
     metacore,
     metatools,
     pharmaversesdtm,
-    admiral,
-    xportr,
+    pharmaverseadam,
+    rtables,
```
Comment (pawelru) [inline DESCRIPTION:19]: This should not be needed as `tern` depends on `rtables` so the latter is automatically attached

Comment (rossfarrugia) [reply DESCRIPTION:19]: i was wondering if not easier to include both for clarity for users to show these are the 2 prominent pharmaverse packages used here? that's how Tad had it before in https://github.com/pharmaverse/end_to_end_examples/blob/main/content/output/demographic/_index.Rmd. i was trying to balance efficiency with clarity for completely brand new users. let me know if you prefer i remove rtables though?

Comment (pawelru) [reply DESCRIPTION:19]: Note that this file is for GitHub Actions only to install all required dependencies - I don't expect users will be looking into this...
If we want to _expose_ this package more then we would have to add additional `library(rtables)` calls, package prefix `rtables::foo` and maybe docs. But this has also a downside - people might think that they would need to load those two libs whereas only one load is required. That's the main reason why we decided to put in into Depends - to have a single library call for simplicity.
Taking into account the goal for this website - I would personally go with simplicity and avoid unnecessary calls to keep the code _small_.

Comment (rossfarrugia) [reply DESCRIPTION:19]: ok sure, makes sense. will update

**Review by rossfarrugia — APPROVED** [review]

**Comment by rossfarrugia** [comment]
@pawelru are you ok with my updates now? i was going to leave review comments for you but i had some time this morning so just dived in and made them to save you effort. main pieces i wanted to add were more descriptive text and sectioning as per the other examples, and i used pharmaverseadam test data. i want the other examples to do similar so we all use same test data across the examples here. let me know and i'll merge if so.

File: tlg/adverse_events.qmd [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  df_explicit_na()
+```
+
+## Start preprocessing
+
+Now we will add some pre-processing to add labels ready for display in the table and how the output will be split.
+
+```{r preproc}
+
+adae <- adae %>%
```
Comment (pawelru) [inline tlg/adverse_events.qmd:39]: rm blank line
```suggestion
```{r preproc}
adae <- adae %>%
```

**Comment by pawelru** [comment]
This is awesome! Thank you very much. I will merge this.

**Pull request merged**